src/xopen/__init__.py

```diff
@@ -499,7 +527,16 @@ def _open_stdin_or_out(mode: str) -> IO:
     return open(std.fileno(), mode=mode, closefd=False)
 
 
-def _open_bz2(filename, mode: str) -> IO:
+def _open_bz2(filename, mode: str, compresslevel, threads):
```

Comment:
I’m not sure that `compresslevel` should actually be handled, at least not in this way. When opening a bz2 file through `bz2.open`, we ignore the `compresslevel` at the moment.

The compression level for Bzip2 files is on quite a different scale than for gzip files. bzip2 and pbzip2 compress at level 9 (block size 900k) by default, and reducing it to 1 (block size 100k) changes runtime and compressed size only a bit (14% slower, 7% bigger in a short test I did, difference in runtime is a bit bigger than that for `pbzip2`).

Since `xopen()` can handle multiple file formats, it may be necessary that it gets another parameter for just the bzip2 compression level if we want full flexibility. But the easiest thing to do for the moment IMO would be to just behave the same as before and just *not* set the compression level.

Reply:
I removed the parsing of compresslevel and added the type hint!